zerver/tests/test_realm.py

```diff
@@ -1207,11 +1215,55 @@ def do_test_realm_update_api(self, name: str) -> None:
         realm = self.update_with_api(name, vals[0])
         self.assertEqual(getattr(realm, name), vals[0])
 
+    def do_test_realm_permission_group_setting_update_api(self, name: str) -> None:
+        realm = get_realm("zulip")
+        setting_group_id_name = name + "_id"
+
+        all_system_user_groups = UserGroup.objects.filter(
+            realm=realm,
+            is_system_group=True,
+        )
+
+        setting_permission_configuration = Realm.REALM_PERMISSION_GROUP_SETTINGS[name]
+
+        default_group_name = setting_permission_configuration.default_group_name
+        default_group = all_system_user_groups.get(name=default_group_name)
+
+        self.set_up_db(name, default_group)
+
+        for user_group in all_system_user_groups:
+            if (
+                (
+                    user_group.name == UserGroup.EVERYONE_ON_INTERNET_GROUP_NAME
+                    and not setting_permission_configuration.allow_internet_group
+                )
+                or (
+                    user_group.name == UserGroup.NOBODY_GROUP_NAME
+                    and not setting_permission_configuration.allow_nobody_group
+                )
+                or (
+                    user_group.name == UserGroup.OWNERS_GROUP_NAME
+                    and not setting_permission_configuration.allow_owners_group
+                )
+                or user_group.name == default_group_name
+            ):
+                continue
```

Comment:
I think we should also test the cases where a user tries to set the setting to these groups and check that these raise an error.